src/daemon/bitcoind/poller.rs

```diff
@@ -911,18 +911,27 @@ fn unvault_spender(
     }
 
     // Second, check if it was spent by an UnvaultEmergency if we are able to, it's as cheap.
-    if let Some(unemer_txid) = unemer_txid(revaultd, &vault)? {
+    let unemer_txid = unemer_txid(revaultd, &vault)?;
+    if let Some(unemer_txid) = unemer_txid {
         if bitcoind.is_current(&unemer_txid)? {
             return Ok(Some(UnvaultSpender::Emergency(unemer_txid)));
         }
     }
 
-    // Finally, fetch the spending transaction and assume it's a Spend.
-    if let Some(spend_txid) = bitcoind.get_spender_txid(&unvault_outpoint, &previous_tip.hash)? {
+    // Finally, fetch the spending transaction
+    if let Some(spender_txid) = bitcoind.get_spender_txid(&unvault_outpoint, &previous_tip.hash)? {
         // FIXME: be smarter, all the information are in the previous call, no need for a
         // second one.
-        if bitcoind.is_current(&spend_txid)? {
-            return Ok(Some(UnvaultSpender::Spend(spend_txid)));
+
+        // Let's double-check that we didn't fetch the cancel, nor the unemer
```

Comment:
Can you be more precise as to why it makes sense? So that when we revisit we get it first glance.

> In theory (read edge cases), the Cancel and UnEmer could have not been current at the last `bitcoind` poll but could be now. Be sure to not wrongly mark a Cancel or UnEmer as a Spend!